src/snowflake/snowpark/_internal/data_source/dbms_dialects/oracledb_dialect.py

```diff
@@ -1,28 +1,33 @@
 #
 # Copyright (c) 2012-2025 Snowflake Computing Inc. All rights reserved.
 #
+from typing import List
+
 from snowflake.snowpark._internal.data_source.dbms_dialects import BaseDialect
+from snowflake.snowpark._internal.utils import quote_name
 from snowflake.snowpark.types import StructType, TimestampType, TimestampTimeZone
 
 
 class OracledbDialect(BaseDialect):
-    def generate_select_query(self, table_or_query: str, schema: StructType) -> str:
+    def generate_select_query(
+        self, table_or_query: str, schema: StructType, raw_schema: List[tuple]
+    ) -> str:
         cols = []
-        for field in schema.fields:
+        for field, raw_field in zip(schema.fields, raw_schema):
             if (
                 isinstance(field.datatype, TimestampType)
                 and field.datatype.tz == TimestampTimeZone.TZ
             ):
                 cols.append(
-                    f"""TO_CHAR({field.name}, 'YYYY-MM-DD HH24:MI:SS.FF9 TZHTZM')"""
+                    f"""TO_CHAR({quote_name(raw_field[0], keep_case=True)}, 'YYYY-MM-DD HH24:MI:SS.FF9 TZHTZM')"""
```

Comment:
have you tested whether oracledb allows quote in identifier? and if it's supported does this approach work?

Reply:
yeah they support double quotes in identifier, and this is covered by the test I added